src/clipping/weileratherton.jl

```diff
@@ -0,0 +1,397 @@
+# Licensed under the MIT License. See LICENSE in the project root.
+# ------------------------------------------------------------------
+
+"""
+    WeilerAtherton()
+
+The Wieler-Atherton algorithm for clipping polygons.
+
+## References
+
+* Weiler, K., & Atherton, P. 1977. [Hidden surface removal using polygon area sorting]
+  (https://dl.acm.org/doi/pdf/10.1145/563858.563896)
+"""
+
+struct WeilerAtherton <: ClippingMethod end
+
+# VertexType has three different types of vertices used for deciding how to navigate the data
+# structure when collecting the polygon rings after clipping.
+abstract type VertexType end
+abstract type Normal <: VertexType end
+abstract type Entering <: VertexType end
+abstract type Exiting <: VertexType end
+
+# Data structure for clipping the polygons. Fields left and right are used depending on the
+# VertexType. If Normal, left points to the following vertex of the original ring, and right
+# to the next  intersection vertex on the edge between point and left.point, or the following
+# original ring vertex if no intersections on the edge. For Entering and Exiting types, left
+# poitns to the following vertex on the clipping ring and right to the following vertex on one
+# of the clipped rings.
+mutable struct RingVertex{VT<:VertexType,M<:Manifold,C<:CRS}
+  point::Point{M,C}
+  left::RingVertex{VT,M,C} where {VT<:VertexType}
+  right::RingVertex{VT,M,C} where {VT<:VertexType}
+
+  function RingVertex{VT,M,C}(point::Point{M,C}) where {VT<:VertexType,M<:Manifold,C<:CRS}
+    v = new(point)
+    v.left = v
+    v.right = v
+    return v
+  end
+end
+
+RingVertex{VT}(point::Point{M,C}) where {VT<:VertexType,M<:Manifold,C<:CRS} =
+  RingVertex{VT,manifold(point),typeof(point.coords)}(point)
+
+function append_vertices!(v1::RingVertex{Normal}, v2::RingVertex{Normal})
+  v2.left = v1.left
+  v1.left = v2
+  v2.right = v1.right
+  v1.right = v2
+end
+
+function get_traversal_ring(ring::Ring)
+  vs = vertices(ring)
+  start = RingVertex{Normal}(vs[1])
+
+  for v in vs[2:end]
+    new = RingVertex{Normal}(v)
+    append_vertices!(start, new)
+    start = new
+  end
+
+  return start.left
+end
+
+is_normal(_::RingVertex{Normal}) = true
+is_normal(_::RingVertex) = false
+
+function insert_intersection!(segment_start::RingVertex, intersection::RingVertex, side::Symbol)
+  segment_end = segment_start.left
+  vertex = segment_start
+
+  l_new = measure(Segment(segment_start.point, intersection.point))
+  while true
+    origin_side = is_normal(vertex) ? :right : side
+    l_current = measure(Segment(segment_start.point, getfield(vertex, origin_side).point))
+    if (l_new < l_current) || (getfield(vertex, origin_side) == segment_end)
+      next_vertex = getfield(vertex, origin_side)
+      setfield!(intersection, side, next_vertex)
+      setfield!(vertex, origin_side, intersection)
+      break
+    end
+    vertex = getfield(vertex, origin_side)
+  end
+end
```

Comment:
Please follow the code style. For example, `segment_start` can be replaced by `start`. You can then add comments to the function body to explain what is happening.

Reply:
Tried to simplify variable namings everywhere.